packages/cli-kit/src/private/node/ui/components/SelectInput.tsx

```diff
@@ -261,52 +262,61 @@ function SelectInputInner<T>(
     )
   } else {
     const optionsHeight = initialItems.length + maximumLinesLostToGroups(initialItems)
+    const minHeight = hasAnyGroup ? 5 : 2
+    const sectionHeight = Math.max(minHeight, Math.min(availableLines, optionsHeight))
     const maxKeyLength = itemsWithKeys
       .map((item) => item.key?.length ?? 0)
       .reduce((lenA, lenB) => Math.max(lenA, lenB), 0)
-    const minHeight = hasAnyGroup ? 5 : 2
-    return (
-      <Box flexDirection="column" ref={ref}>
-        <Box
-          flexDirection="column"
-          height={Math.max(minHeight, Math.min(availableLines, optionsHeight))}
-          overflowY="hidden"
-        >
-          {state.visibleOptions.map((item: ItemWithKey<T>, index: number) => (
-            <Item
-              key={item.key}
-              item={item}
-              previousItem={state.visibleOptions[index - 1]}
-              highlightedTerm={highlightedTerm}
-              isSelected={item.value === state.value}
-              items={state.visibleOptions}
-              enableShortcuts={enableShortcuts}
-              hasAnyGroup={hasAnyGroup}
-              maxKeyLength={maxKeyLength}
-            />
-          ))}
-        </Box>
 
-        {noItems ? (
-          <Box marginTop={1} marginLeft={3} height={2}>
-            <Text dimColor>Try again with a different keyword.</Text>
-          </Box>
-        ) : (
-          <Box marginTop={1} marginLeft={3} flexDirection="column">
-            <Text dimColor>
-              {infoMessage
-                ? infoMessage
-                : `Press ${figures.arrowUp}${figures.arrowDown} arrows to select, enter to confirm.`}
-            </Text>
-            {hasMorePages ? (
-              <Text>
-                <Text bold>1-{items.length} of many</Text>
-                {morePagesMessage ? `  ${morePagesMessage}` : null}
-              </Text>
+    return (
+      <Box flexDirection="row">
+        <Box flexDirection="column" ref={ref}>
+          <Box flexDirection="row" height={sectionHeight} width="100%">
+            <Box flexDirection="column" height={sectionHeight} overflowY="hidden" flexGrow={1}>
+              {state.visibleOptions.map((item: ItemWithKey<T>, index: number) => (
+                <Item
+                  key={item.key}
+                  item={item}
+                  previousItem={state.visibleOptions[index - 1]}
+                  highlightedTerm={highlightedTerm}
+                  isSelected={item.value === state.value}
+                  items={state.visibleOptions}
+                  enableShortcuts={enableShortcuts}
+                  hasAnyGroup={hasAnyGroup}
+                  maxKeyLength={maxKeyLength}
+                />
+              ))}
+            </Box>
+            {hasLimit ? (
+              <Scrollbar
+                containerHeight={sectionHeight}
```

Comment:
Do we need to pass the `containerHeight` as a param to scrollbar? Can't Scrollbar track its own height via a ref? I'd rather have fewer params if possible. This would make it easier to embed the scrollbar in other parts of the UI if needed.